source/code/plugins/in_agent_telemetry.rb

```diff
@@ -57,6 +57,8 @@ def start
     def shutdown
       if defined?(OMS::Configuration.telemetry_interval)
         @finished = true
+        log.debug("in_agent_telemetry: thread state before shutdown '#{@thread.status}'")
+        @thread.exit
```

Comment:
already added in my finally-merged qos-errors pr :)

Reply:
Great then I will remove this. :+1: